core/src/main/java/org/apache/iceberg/avro/ProjectionDatumReader.java

```diff
@@ -60,6 +60,7 @@ public void setRowPositionSupplier(Supplier<Long> posSupplier) {
   @Override
   public void setSchema(Schema newFileSchema) {
     this.fileSchema = newFileSchema;
+    AvroSchemaUtil.convertToDeriveNameMapping(this.fileSchema);
```

Comment:
I think this line should change/go away after addressing some of the other comments.